crates/phactory/src/contracts/pink.rs

```diff
@@ -164,6 +158,12 @@ impl contracts::NativeContract for Pink {
             })?;
         Ok(effects)
     }
+}
+
+impl NativeContractMore for Pink {
+    fn id(&self) -> contracts::NativeContractId {
+        Pink::address_to_id(&self.instance.address).into()
```

Comment:
Confused. So the address and the contract id are different, right? I hope the address within the ink runtime should match the contract id on the blockchain.

Otherwise it will be very confusing. Say we have a an ink contract deployed. Then it's natural to save it's address and allow other contract to call it. If it's within the same group, it's easy. But if we do cross group messaging, then will it have a different id then the one in the ink vm?

Reply:
The formula to generate ink! address is out of our control. Doesn't have an idea on how to make the ink address match it's contract id.
How's the ink contract supposed to expose it's address for cross group messaging? Can we provide it a pink address along side?

Reply:
Submitted a [PR](https://github.com/paritytech/substrate/pull/10521) to substrate. Let's wait for it to be merged.

Reply:
LGTM